### ISSUE TITLE
Workload error recovery problems.

### ISSUE THREAD

**Issue by lewing** [dotnet/sdk#17752]
We've been testing `dotnet workload install` on multiple platforms and we've run into a few cases where a network failure occurred.  If you try to run `dotnet workload install` again after the network error we've seen a different failure that looks like (on windows)

```
Writing workload pack installation record for Microsoft.NET.Runtime.Emscripten.Python version 6.0.0-preview.6.21268.1...
Installing pack Microsoft.NET.Runtime.Emscripten.Sdk version 6.0.0-preview.6.21268.1...
Workload installation failed, rolling back installed packs...
Installing workload manifest microsoft.net.sdk.android version 11.0.200-ci.main.256.
Installation roll back failed: Failed to install manifest microsoft.net.sdk.android version 11.0.200-ci.main.256: The transaction has aborted..
Rolling back pack Microsoft.NETCore.App.Runtime.Mono.browser-wasm installation...
Rolling back pack Microsoft.NET.Runtime.MonoAOTCompiler.Task installation...
Rolling back pack Microsoft.NET.Runtime.WebAssembly.Sdk installation...
Rolling back pack Microsoft.NETCore.App.Runtime.AOT.Cross.browser-wasm installation...
Rolling back pack Microsoft.NET.Runtime.Emscripten.Node installation...
Rolling back pack Microsoft.NET.Runtime.Emscripten.Python installation...
Rolling back pack Microsoft.NET.Runtime.Emscripten.Sdk installation...
Workload installation failed: One or more errors occurred. (Number of entries expected in End Of Central Directory does not correspond to number of entries in Central Directory.)
```

And exits

Running install a third time does seem to succeed after that.

**Comment by lewing** [comment]
cc @pavelsavara @radekdoulik

**Comment by sfoslund** [comment]
@lewing are you still seeing this? Can you provide a repro version?

**Comment by sfoslund** [comment]
Ping @lewing, is this still an issue?

**Comment by lewing** [comment]
@sfoslund we're still seeing rollback problems in preview5 https://github.com/dotnet/runtime/issues/54342

**Comment by sfoslund** [comment]
I've filed https://github.com/dotnet/sdk/issues/18450 for the installation issues in that issue. That error seems to be different from the one you originally reported here though, are you still seeing the `One or more errors occurred. (Number of entries expected in End Of Central Directory does not correspond to number of entries in Central Directory.)` you mentioned above?

**Comment by lewing** [comment]
haven't seen that recently

**Comment by sfoslund** [comment]
Sounds good, I'll close this issue then and track the other error in #18450

**Comment by doniyorniyozov** [comment]
I Still have this issue when doing "dotnet workload install maui"

**Comment by almas-f** [comment]
> I Still have this issue when doing "dotnet workload install maui"

I have the same issue while installing maui workload

**Comment by sfoslund** [comment]
@doniyorniyozov @almas-f which errors have you been seeing, can you provide the output from the command? What versions of .NET are you using?

**Comment by doniyorniyozov** [comment]
@sfoslund I'm getting "Workload with id maui is not recognized." message. I tried to repair and it didn't work either. How I can completely delete all workloads and reinstall again? 
![image](https://user-images.githubusercontent.com/37545941/129646729-10dcf89d-2f8c-454f-b541-a19da4e44ff7.png)

**Comment by sfoslund** [comment]
@doniyorniyozov what's the output of dotnet workload search? What OS are you on?

**Comment by doniyorniyozov** [comment]
@sfoslund 
OS: Windows.

This is what I get when I do dotnet workload search. 

![image](https://user-images.githubusercontent.com/37545941/129763041-028e3d0f-6811-43b1-903a-d8c95234c8ee.png)

**Comment by sfoslund** [comment]
Okay, based on the output of workload search it looks like you're missing manifests. Does the dotnet/sdk-manifests dir exist? If so, what's inside of it? If not, then your sdk installation is broken and you'll want to either repair or re-install the sdk.

**Comment by pavelsavara** [comment]
I saw `Workload installation failed: Number of entries expected in End Of Central Directory does not correspond to number of entries in Central Directory` on my windows machine today too with `7.0.0-alpha.1.21560.2`. It was in single purpose `runtime\artifacts\bin\dotnet-workload` folder. 
Possibly I caused it myself by interrupting the workload installation at wrong moment and then restarted it. 

I resolved it by deleting whole dotnet folder and installing again. Then it just worked.
